django/template/library.py

```diff
@@ -153,6 +153,92 @@ def compile_func(parser, token):
         else:
             raise ValueError("Invalid arguments provided to simple_tag")
 
+    def simple_block_tag(self, func=None, takes_context=None, name=None):
+        """
+        Register a callable as a compiled block template tag. Example:
+
+        @register.simple_block_tag
+        def hello(content):
+            return 'world'
+        """
+
+        def dec(func):
+            (
+                params,
+                varargs,
+                varkw,
+                defaults,
+                kwonly,
+                kwonly_defaults,
+                _,
+            ) = getfullargspec(unwrap(func))
+            function_name = name or func.__name__
+
+            @wraps(func)
+            def compile_func(parser, token):
+                tag_params = params.copy()
+
+                if takes_context:
+                    if len(tag_params) >= 2 and tag_params[1] == "content":
+                        del tag_params[1]
+                    else:
+                        raise TemplateSyntaxError(
+                            f"'{function_name}' is decorated with takes_context=True so"
+                            " it must have a first argument of 'context' and a second "
+                            "argument of 'content'"
+                        )
+                elif tag_params and tag_params[0] == "content":
+                    del tag_params[0]
+                else:
+                    raise TemplateSyntaxError(
+                        f"'{function_name}' must have a first argument of 'content'"
+                    )
+
+                bits = token.split_contents()[1:]
+                target_var = None
+                if len(bits) >= 2 and bits[-2] == "as":
+                    target_var = bits[-1]
+                    bits = bits[:-2]
+
+                # Use underscore to separate "end" to match user styling
+                end_tag_name = (
+                    f"end_{function_name}"
+                    if "_" in function_name
+                    else f"end{function_name}"
+                )
```

Comment:
In hindsight, it would be surprising. I think personally I prefer the naming consistency. I've made it customizable now, which lets people style it however they need (with tests, of course).

Reply:
Does this "end tag name" customization is necessary? I would suggest we remove it and add it if eventually some asks for it...